### PR TITLE
fix: update root node branch name on changing base branch [IDE-841]

### PR DESCRIPTION
**Pull request by DariusZdroba**
### Description

Root Node base branch text was only initialized never updated, this PR updates it whenever the other RootTreeNodes update.
Note: The state is and was already saved as intended in the folder config.
### Checklist

- [ ] Tests added and all succeed
- [ ] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt [hunk 1]
```diff
@@ -552,8 +552,21 @@ class SnykToolWindowPanel(
 
         val newContainerTreeNodeText = getNewContainerTreeNodeText(settings, containerResultsCount, addHMLPostfix)
         newContainerTreeNodeText?.let { rootContainerIssuesTreeNode.userObject = it }
+
+        val newRootTreeNodeText = getNewRootTreeNodeText()
+        newRootTreeNodeText.let { rootTreeNode.info = it }
```
Comment (DariusZdroba) [inline src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:557]: Can this be part of the `updateTreeRootNodesPresentation` or is the context for this only supposed to extend the product issue nodes? 

My intended logic was to have all the RootNodes update at the same time when this is triggered.

**Comment by bastiandoetsch** [comment]
Would you please update the changelog?

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt (hunk 1 above)
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:557]: This is fine, I think.

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt [hunk 2]
```diff
@@ -552,8 +552,21 @@ class SnykToolWindowPanel(
...
     }
 
+    private fun getNewRootTreeNodeText() : String {
+        val folderConfig = service<FolderConfigSettings>().getFolderConfig(project.basePath.toString())
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:561]: You shouldn't use the `project.basePath` as stated in the IntelliJ SDK docs. In the future we'll most likely use content root nodes like in Eclipse in the future. We can pair on this to explore how to solve this issue, but I feel like using `project.basePath` with folderConfigs creates inconsistencies.

Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:561]: That said, it's like that in the other code, too. Just wondering if we can improve it.

Comment (DariusZdroba) [reply src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:561]: Ok, as discussed this is part of a bigger issue, let's pair on this sometime and figure out all the instances of the `project.basePath` that we use where we should use the content root.

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**